view.c

```diff
@@ -55,17 +55,21 @@ view_activate(struct cg_view *view, bool activate)
 static bool
 view_extends_output_layout(struct cg_view *view, struct wlr_box *layout_box)
 {
-	int width, height;
-	view->impl->get_geometry(view, &width, &height);
+	struct wlr_box view_box;
+	view->impl->get_geometry(view, &view_box);
 
-	return (layout_box->height < height || layout_box->width < width);
+	return (layout_box->height < view_box.height || layout_box->width < view_box.width);
 }
 
 static void
 view_maximize(struct cg_view *view, struct wlr_box *layout_box)
 {
-	view->lx = layout_box->x;
-	view->ly = layout_box->y;
+	struct wlr_box view_box;
+	view->impl->get_geometry(view, &view_box);
+
+	// Do not forget to adjust position according to top left corner declared in view geometry
+	view->lx = layout_box->x - view_box.x;
+	view->ly = layout_box->y - view_box.y;
```

Comment:
Hm, I'm not sure this is the correct spot to read the window geometry. The window geometry is likely to change when the client redraws in the maximized state. This function runs before though, when we send the event to ask the client to redraw itself maximized.

Reply:
@emersion, I understand the client may adjust geometry using `xdg_surface::set_window_geometry()` before calling (again) `wl_surface::commit()`. wlroots then calls the handler for map event and in Cage `view_map()`, `view_position()` and finally `view_maximize()` or `view_center()` where the new geometry can be accessed.